config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -0,0 +1,280 @@
+        error_exit "Failed to destroy the libvirt pool"
+    
+    virsh -d 0 -c $LIBVIRT_URI pool-undefine $LIBVIRT_POOL ||
+        error_exit "Failed to undefine the libvirt pool"
+
+    echo "Deleted libvirt image successfully"
+
+    # Remove the libvirt_volume_name from peer-pods-cm configmap
+    delete_libvirt_vol_from_peer_pods_cm
+
+}
+
+# Function to delete the libvirt image id  from the peer-pods-cm configmap
+
+function delete_libvirt_vol_from_peer_pods_cm() {
+    echo "Deleting libvirt volume from peer-pods-cm configmap"
+
+    # Check if the peer-pods-cm configmap exists
+    if ! kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator >/dev/null 2>&1; then
+        echo "peer-pods-cm configmap does not exist. Skipping deleting the ami id"
+        return
+    fi
+
+    # Delete the libvirt image id from peer-pods-cm configmap
+    kubectl patch configmap peer-pods-cm -n openshift-sandboxed-containers-operator \
+    --type merge -p "{\"data\":{\"LIBVIRT_IMAGE_ID\":\"\"}}" ||
+        error_exit "Failed to delete the libvirt image id from peer-pods-cm configmap"
+    echo "libvirt image id deleted from peer-pods-cm configmap successfully"
+}
+
+# display help message
+
+function display_help() {
+    echo "This script is used to create libvirt qcow2 for podvm"
+    echo "Usage: $0 [-c|-C] [-- install_binaries]"
+    echo "Options:"
+    echo "-c  Create image"
+    echo "-C  Delete image"
+}
+
+function install_packages(){
+
+    install_binary_packages
+
+    #Install other libvirt specific binaries packages
+    export ARCH=$(uname -m)
+    if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then
+        subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY}
+    fi
+    subscription-manager repos --enable codeready-builder-for-rhel-9-${ARCH}-rpms
+    dnf install -y gcc genisoimage qemu-kvm libvirt-client
+
+    GO_VERSION="1.21.9"
+    curl https://dl.google.com/go/go${GO_VERSION}.linux-${ARCH/x86_64/amd64}.tar.gz -o go${GO_VERSION}.linux-${ARCH/x86_64/amd64}.tar.gz && \
+    rm -rf /usr/local/go && tar -C /usr/local -xzf go${GO_VERSION}.linux-${ARCH/x86_64/amd64}.tar.gz && \
+    rm -f go${GO_VERSION}.linux-${ARCH/x86_64/amd64}.tar.gz
+
+    export PATH="/usr/local/go/bin:${PATH}"
+    export GOPATH="/src"
+
+# Install packer. Packer doesn't does not have prebuilt s390x arch binaries above Packer version 0.1.5 
+    PACKER_VERSION="v1.9.4"
+    if [ "${ARCH}" == "s390x" ]; then
+        git clone --depth 1 --single-branch https://github.com/hashicorp/packer.git -b ${PACKER_VERSION}
+        cd packer
+        sed -i -- "s/ALL_XC_ARCH=.*/ALL_XC_ARCH=\"${ARCH}\"/g" scripts/build.sh
+	    sed -i -- "s/ALL_XC_OS=.*/ALL_XC_OS=\"Linux\"/g" scripts/build.sh
+        make bin && cp bin/packer /usr/local/bin/
+        cd $OLDPWD
+    else
+        yum-config-manager --add-repo https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo && \
```

Comment:
packer is already being installed for x86_64 as part of install_binary_packages. So this is not needed here

Reply:
You are already building packer here for s390x. When this code runs on x86_64, then the original install_binary_packages (which you call in install_packages) should install the packer binary. 
I don't understand why the hashicorp repo needs to be added here  ? What am I missing ?

Reply:
Got your point. No need to hashicorp repo. Added condition to install additional packages for s390x only